operator/controllers/loki/lokistack_dashboards_controller.go

```diff
@@ -0,0 +1,84 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/ViaQ/logerr/v2/kverrors"
+	"github.com/go-logr/logr"
+	lokiv1 "github.com/grafana/loki/operator/apis/loki/v1"
+	"github.com/grafana/loki/operator/internal/manifests/openshift"
+	"k8s.io/apimachinery/pkg/labels"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+var deletesOnlyPred = builder.WithPredicates(predicate.Funcs{
+	UpdateFunc:  func(e event.UpdateEvent) bool { return false },
+	CreateFunc:  func(e event.CreateEvent) bool { return false },
+	DeleteFunc:  func(e event.DeleteEvent) bool { return true },
+	GenericFunc: func(e event.GenericEvent) bool { return false },
+})
+
+// LokiStackDasboardsReconciler cleans up all remaining
+// dashboards when all LokiStack objects are removed.
+type LokiStackDasboardsReconciler struct {
+	client.Client
+	Log logr.Logger
+}
+
+// Reconcile removes all LokiStack dashboard ConfigMap objects on OpenShift clusters when
+// the last LokiStack custom resource is deleted.
```

Comment:
nit: I would suggest this comment be moved to either line 56 or 69. This way it would live closer to the actual part of the code that is referring to in case this ever gets changed.